src/python/ServerUtilities.py

```diff
@@ -880,3 +880,34 @@ def downloadFromS3ViaPSU(filepath=None, preSignedUrl=None, logger=None):
         raise Exception("Download failure with %s. File %s was not created" % (downloadCommand, filepath))
 
     return
+
+
+class MeasureTime:
+    """
+    Context manager to measure how long a portion of code takes. 
+    It is intended to be used as:
```

Comment:
remember to update the doc string. BTW, code in RESTCache reads much better now, thx

Reply:
thanks stefano!